tests/python/pants_test/engine/test_scheduler.py

```diff
@@ -100,6 +171,35 @@ def test_transitive_params(self):
     # we're just testing transitively resolving products in this file.
     self.assertTrue(isinstance(result_d, D))
 
+  @contextmanager
+  def _assert_execution_error(self, expected_msg):
+    # TODO(#7303): use self.assertRaisesWithMessageContaining()!
+    with self.assertRaises(ExecutionError) as cm:
```

Comment:
Could this be `self.assertRaisesRegexp`?

Reply:
It might be, but I'm realizing now that this probably would be difficult to convert to `self.assertRaisesWithMessageContaining()` anyway because of the way it preprocesses the exception message. That could be converted to a regexp match (which I tried to do at one point), but I think stripping the locations from the traceback is much more robust and was just going to remove this comment, actually.